console/v1/types.go

```diff
@@ -0,0 +1,10 @@
+package v1
+
+// Represents a standard link that could be generated in HTML
+type Link struct {
+	// text is the display text for the link
+	Text string `json:"text" protobuf:"bytes,1,opt,name=text"`
+	// href is the absolute secure URL for the link (must use https)
+	// +kubebuilder:validation:Pattern=^$|^((https):\/\/?)[^\s()<>]+(?:\([\w\d]+\)|([^[:punct:]\s]|\/?))$
```

Comment:
Do we need this: `+kubebuilder:validation:Pattern`

Reply:
Comparing to:
- https://github.com/openshift/api/blob/master/config/v1/types_console.go#L55
- https://github.com/openshift/cluster-config-operator/blob/master/manifests/0000_10_config-operator_01_console.crd.yaml#L50

For our top level config, we apparently duplicate the validation in both the API object & CRD.

Do we need to do that here as well?  Related to:
- https://github.com/openshift/console-operator/pull/229/files#diff-46afeb1a2b75c5fd69541450a98218b4R62
- https://github.com/openshift/console-operator/pull/229/files#diff-ab08ffa2ed478ebd4e5546efac3e9c54R53
- https://github.com/openshift/console-operator/pull/229/files#diff-da63edd9c81dbcd28e5975f8111dfa97R59

@zherman0

Reply:
eliminated.